File: Data Structures & Algorithms/duplicate-integer/submission-3.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        seen = []
        if len(nums) >1:
            for num in nums:
                if num in seen:
                    return True
                    del seen

                seen.append(num)

        del seen
        return False
        

File: Data Structures & Algorithms/binary-search/submission-3.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {

        int hi = nums.size() -1;
        int lo = 0;
        int mid = 0;

        while(hi>=lo)
        {
            mid = lo + (hi - lo)/2;
            
            if(nums[mid] == target) return mid;

            if(nums[mid] < target)
            {
                lo = mid +1;
            }
            else
            {
                hi = mid -1;
            }
        }
        return nums[mid] == target ? mid : -1;
        
    }
};
